core/embed/rust/src/ui/model_mercury/layout.rs

```diff
@@ -581,11 +655,33 @@ extern "C" fn new_confirm_summary(n_args: usize, args: *const Obj, kwargs: *mut
         let cancel_text: Option<TString> =
             kwargs.get(Qstr::MP_QSTR_cancel_text)?.try_into_option()?;
 
-        let flow = flow::confirm_summary::new_confirm_summary(
-            title,
-            items,
-            account_items,
-            fee_items,
+        let mut summary_params = ShowInfoParams::new(title.clone())
+            .with_menu_button()
+            .with_footer(TR::instructions__swipe_up.into(), None)
+            .with_swipe_up();
+        for pair in IterBuf::new().try_iterate(items)? {
+            let [label, value]: [TString; 2] = util::iter_into_array(pair)?;
+            summary_params = unwrap!(summary_params.add(label, value));
+        }
+
+        let mut account_params =
+            ShowInfoParams::new(TR::send__send_from.into()).with_cancel_button();
+        for pair in IterBuf::new().try_iterate(account_items)? {
```

Comment:
Can't this be extracted into a `ShowInfoParams::add_items`?

Reply:
Not without making `ShowInfoParams` dependent on `micropython` crate.